include/criterion/assert.h

```diff
@@ -30,109 +35,1041 @@
 
 # include "internal/assert.h"
 
-// Base assertions
+/**
+ * @defgroup BaseAsserts Base assertions
+ * @{
+ */
+
+/**
+ * Fails always.
+ *
+ * Call: \c cr_assert_fail([FormatString, [Args...]]).\n
+ * The test is marked as failure and the execution of the function is aborted.
+ * The optional string is printed on failure.
+ *
+ * @param[in] fmt printf like format string
+ *
```

Comment:
All macros should probably have an `@param[in] Args... the format paramaters` or something.